pkgs/by-name/dc/dc3dd/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchzip
+, perl538Packages
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dc3dd";
+  version = "7.3.1";
+
+  src = fetchzip {
+      url = "http://downloads.sf.net/sourceforge/dc3dd/dc3dd-${finalAttrs.version}.zip";
+      hash = "sha256-SYDoqGlsROHX1a0jJX11F+yp6CeFK+tZbYOOnScC6Ig=";
+  };
+
+  postPatch = ''
+    chmod +x configure
+  '';
+
+  buildInputs = [ perl538Packages.LocaleGettext ];
+
+  buildPhase = ''
+    runHook preBuild
+    if stdenv.hostPlatform.isi686 && stdenv.hostPlatform.isx86_64
+    then
+      ./configure --prefix=$out
+    else
+      ./configure --build=arm --prefix=$out
```

Comment:
Have you understood how the things work in Nixpkgs?

Reply:
For this package I'm using existing nix pkgs examples. The reported code works correctly in the `buildPhase`. But if you have better approach, I'm here to listen.

Reply:
At first glances could `configurePlatforms` (if needed combined with `configureFlags`) help  to keep the arch selection simpler.

Reply:
I can't remember a single example of `if stdenv.hostPlatform.isi686 && stdenv.hostPlatform.isx86_64` in a configure phase in Nixpkgs.

Reply:
Fixed. I tried to use `configureFlags = [ "--prefix=$out" ];` but during the compiling it didn't generate the files inside the `result` folder during my tests, so I chose to not use it.